@here/harp-mapview/lib/DecodedTileHelpers.ts

```diff
@@ -415,6 +414,103 @@ export function getMaterialConstructor(technique: Technique): MaterialConstructo
     }
 }
 
+/**
+ * Calculates numerical value of `Value | Expr`.
+ *
+ * Function takes care about color interpolation (when @param zoom is set) as also parsing
+ * string encoded colors.
+ *
+ * Returns `defaultValue` if evaluation result is not `number` or is `NaN`.
+ *
+ * @param value the value of property defined in technique
+ * @param defaultValue return value used if `value` yields invalid result
+ * @param zoomLevel zoom level used for interpolation.
+ * @returns value if property evaluates to `number`, else `defaultValue`
+ */
+export function getNumberPropertyValueSafe<D = number>(
+    property: Value | Expr | undefined,
+    defaultValue: D,
+    envOrLevel: number | Env,
+    pixelToMeters: number = 1.0
+): number | D {
+    if (property === undefined || property === null) {
+        return defaultValue;
+    }
+    const r = getPropertyValue(property, envOrLevel as number, pixelToMeters);
+    return typeof r !== "number" || isNaN(r) ? defaultValue : r;
+}
+
+/**
+ * Calculates the numerical color value of the `Value | Expr` into TTRRGGAA format.
+ *
+ * Function takes care about color interpolation (when @param zoom is set) as also parsing
+ * string encoded colors.
+ *
+ * Evaulates [[Expr]] instances using [[getPropertyValue]] and parses string using
+ * [[parseStringEncodedColor]].
+ *
+ * @param value the value of color property defined in technique
+ * @param defaultValue return value used if `value` yields invalid result
+ * @param zoomLevel zoom level used for interpolation.
+ * @returns color as `number` if property evaluates to proper color, else `defaultValue`
+ */
+export function getColorPropertyValueSafe<D = number>(
+    property: Value | Expr | undefined,
+    defaultValue: D,
+    envOrLevel: number | Env
+): number | D {
+    if (property === undefined || property === null) {
+        return defaultValue;
+    }
+    const evaluated = getPropertyValue(property, envOrLevel as number);
+
+    if (evaluated === undefined || property === null) {
```

Comment:
Shouldn't this be `evaluated === undefined || evaluated === null` ?

Reply:
Fixed